nest-forms-backend/src/nases/utils-services/tokens.nases.service.ts

```diff
@@ -409,48 +431,79 @@ export default class NasesUtilsService {
-                  <SenderId>${senderId}</SenderId>
-                  <RecipientId>${
-                    process.env.NASES_RECIPIENT_URI ?? ''
-                  }</RecipientId>
-                  <MessageType>${pospID}</MessageType>
-                  <MessageSubject>${subject}</MessageSubject>
-                  <Object Id="${form.id}" IsSigned="${
-                    isSigned ? 'true' : 'false'
-                  }" Name="${title}" Description="" Class="FORM" MimeType="${mimeType}" Encoding="${encoding}">${message}</Object>
-    ${attachments}
-                </MessageContainer>
-              </Body>
-            </SKTalkMessage>
-      `
+
+    attachments.push({
+      $: {
+        Id: form.id,
+        IsSigned: isSigned ? 'true' : 'false',
+        Name: title,
+        Description: '',
+        Class: 'FORM',
+        MimeType: mimeType,
+        Encoding: encoding,
+      },
+      // TODO When message should be created as object
+      _: message,
+    })
+
+    const template = {
+      SKTalkMessage: {
+        $: {
+          xmlns: 'http://gov.sk/SKTalkMessage',
+          'xmlns:xsi': 'http://www.w3.org/2001/XMLSchema-instance',
+          'xmlns:xsd': 'http://www.w3.org/2001/XMLSchema',
+        },
+        EnvelopeVersion: '3.0',
+        Header: {
+          MessageInfo: {
+            Class: 'EGOV_APPLICATION',
+            PospID: pospID,
+            PospVersion: pospVersion,
+            MessageID: form.id,
+            CorrelationID: correlationId,
+          },
+        },
+        Body: {
+          MessageContainer: {
+            $: {
+              xmlns: 'http://schemas.gov.sk/core/MessageContainer/1.0',
+            },
+            MessageId: form.id,
+            SenderId: senderId,
+            RecipientId: process.env.NASES_RECIPIENT_URI ?? '',
+            MessageType: pospID,
+            MessageSubject: subject,
+            Object: attachments,
+          },
+        },
+      },
+    }
     // <Object Id="${data.id}" IsSigned="${String(data.isSigned)}" Name="${data.formName}" Description="${data.fromDescription}" Class="FORM" MimeType="application/vnd.etsi.asic-e+zip" Encoding="Base64">
     // <Object Id="${data.id}" IsSigned="${String(data.isSigned)}" Name="${data.formName}" Description="${data.fromDescription}" Class="FORM" MimeType="application/x-eform-xml" Encoding="XML">
-    envelope = this.formatXmlToOneLine(envelope)
-    return envelope
+
+    // We have already (safe?) xml we want to pase in and xml2js can't do that.
+    // So we paste it as CDATA and remove tags for the same result
+    const builder = new Builder({
+      // eslint-disable-next-line unicorn/text-encoding-identifier-case
+      xmldec: { version: '1.0', encoding: 'utf-8' },
+      cdata: true,
+      renderOpts: {
+        pretty: false,
+      },
+    })
+    const envelope = builder.buildObject(template)
+    // TODO removing of CDATA tag is unsafe
+    return this.formatXmlToOneLine(envelope)
```

Comment:
Can we remove `formatXmlToOneLine` entirely? One of the reasons of this tasks is get rid of these string manipulations. I see there's `pretty: false` so it should be already formatted in one line.

Reply:
This was left as a helper before I implemented your solution - the problematic part was in a string with formatting. Deleted in the next commit.